src/components/MonitorListItem.vue

```diff
@@ -187,6 +199,101 @@ export default {
 
             window.localStorage.setItem("monitorCollapsed", JSON.stringify(storageObject));
         },
+        /**
+         * Initializes the drag operation if the monitor is draggable.
+         * Prevents dragging of group monitors.
+         * @param {DragEvent} event - The dragstart event triggered by the browser.
+         * @returns {void} This method does not return anything.
+         */
+        onDragStart(event) {
+            // Only allow dragging non-group monitors (so groups act as drop targets)
+            if (this.monitor.type === "group") {
+                event.preventDefault();
+                return;
+            }
+
+            try {
+                event.dataTransfer.setData("text/monitor-id", String(this.monitor.id));
+                event.dataTransfer.effectAllowed = "move";
+                this.dragging = true;
+            } catch (e) {
+                // ignore
+            }
+        },
+
+        onDragEnd() {
+            this.dragging = false;
+        },
+
+        onDragEnter(event) {
+            // show visual feedback if this item is a group
+            if (this.monitor.type === "group") {
+                this.dragOver = true;
+            }
+        },
+
+        onDragLeave(event) {
+            if (this.monitor.type === "group") {
+                this.dragOver = false;
+            }
+        },
+
+        async onDrop(event) {
+            event.preventDefault();
+            this.dragOver = false;
+
+            // Only groups accept drops
+            if (this.monitor.type !== "group") {
+                return;
+            }
+
+            const draggedId = event.dataTransfer.getData("text/monitor-id");
+            if (!draggedId) {
+                return;
+            }
+
+            const draggedMonitorId = parseInt(draggedId);
+            if (isNaN(draggedMonitorId) || draggedMonitorId === this.monitor.id) {
+                return;
+            }
+
+            const draggedMonitor = this.$root.monitorList[draggedMonitorId];
+            if (!draggedMonitor) {
+                return;
+            }
+
+            // Save original parent so we can revert locally if server returns error
+            const originalParent = draggedMonitor.parent;
+
+            // Prepare a full monitor object (clone) and set new parent
+            const monitorToSave = JSON.parse(JSON.stringify(draggedMonitor));
```

Comment:
Slightly newer syntax for this which you might not be aware of

```suggestion
            const monitorToSave = structuredClone(draggedMonitor);
```

Reply:
Thanks for the information, I'll adjust that.

Reply:
I tried your suggestion locally, but it doesn't seem to work. I don't quite understand why.